File: 2019/01/part_1.rb
# Usage: From root folder
# $ ruby 2019/01/part_1.rb 2019/01/input

def required_fuel mass
  (mass.to_i/3).floor - 2
end

puts File.open(ARGV[0]).readlines.inject(0){|sum, n| sum + required_fuel(n)}

File: 2015/08/part_1.py
#  Usage: From root folder
#  $ python 2015/08/part_1.py 2015/08/input

import sys

if __name__ == "__main__":
    with open(sys.argv[1]) as file:
        strings = [s.strip() for s in file.readlines()]

    print(sum([len(s) - len(eval(s)) for s in strings]))
